### PR TITLE
feat: prepare ui for image upload

### PR DESCRIPTION
**Pull request by TomasSlama**
[FX-4107]

### Description

Prepare use interface with mocked uploading functionality. After a discussion with Matt, we will reuse FileInput from Picasso

### How to test

- check [temploy](https://picasso.toptal.net/fx-4107-ui/?path=/story/forms-richtexteditor--richtexteditor)

### Screenshots

image
image
image

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests

**Breaking change**

- [x] codemod is created and showcased in the changeset
- [x] test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-4107]: https://toptal-core.atlassian.net/browse/FX-4107?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/picasso-rich-text-editor/src/plugins/ImagePlugin/ImagePluginModal.tsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+import type { FileUpload } from '@toptal/picasso/FileInput'
+
+type Props = {
+  onClose: () => void
+  isOpen: boolean
+}
+
+const MAX_SIZE = 1
+
+// this is just taken from the FileInput example to showcase the progress
```
Comment (TomasSlama) [inline packages/picasso-rich-text-editor/src/plugins/ImagePlugin/ImagePluginModal.tsx:12]: read this comment ☝️, this is just a mock

File: packages/picasso-rich-text-editor/src/plugins/ImagePlugin/ImagePluginModal.tsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+  const handleConfirm = () => {
+    console.log({ files, alt })
+    onClose()
+  }
+
+  const disabled = !files.length || !alt
+
+  return (
+    <Modal onClose={onClose} open={isOpen} size='small'>
```
[the remark this answers is not in the document]
Comment (TomasSlama) [reply packages/picasso-rich-text-editor/src/plugins/ImagePlugin/ImagePluginModal.tsx:103]: yeah, we don't have such size in Picasso, I'm discussing it with Matt

File: packages/picasso-rich-text-editor/src/plugins/ImagePlugin/ImagePluginModal.tsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+const ImagePluginModal = ({ isOpen, onClose }: Props) => {
+  const { files, upload, remove } = useUploader({ maxSize: MAX_SIZE })
+  const [alt, setAlt] = useState('')
+
+  const handleConfirm = () => {
+    console.log({ files, alt })
+    onClose()
+  }
+
+  const disabled = !files.length || !alt
```
[the remark this answers is not in the document]
Comment (TomasSlama) [reply packages/picasso-rich-text-editor/src/plugins/ImagePlugin/ImagePluginModal.tsx:100]: Good question, I asked Matt and can have it as optional

[the remark this answers is not in the document]
Comment (TomasSlama) [reply packages/picasso-rich-text-editor/src/plugins/ImagePlugin/ImagePluginModal.tsx]: done

File: packages/picasso-rich-text-editor/src/plugins/ImagePlugin/ImagePluginModal.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (TomasSlama) [reply packages/picasso-rich-text-editor/src/plugins/ImagePlugin/ImagePluginModal.tsx:103]: @mkrl Matt approved. Lets reuse what we have

**Comment by TomasSlama** [comment]
@toptal-anvil ping reviewers

**Review by mkrl — APPROVED** [review]
Overall looks good for now, but one concern that I have is that the upload state is preserved between modal opening/closing. I think the only purpose should be upload => return URL => close and insert into RTE

But the uploader state seems to persist. I think we can still fix it later in the next PR for the task.

File: packages/picasso-rich-text-editor/src/plugins/ImagePlugin/ImagePluginButton.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import { useRTEPluginContext } from '../api'
+import RichTextEditorButton from '../../RichTextEditorButton'
+
+export type Props = {
+  'data-testid'?: string
+  onClick: () => void
+}
+
+const ImagePluginButton = ({ 'data-testid': testId, onClick }: Props) => {
+  const { disabled, focused } = useRTEPluginContext()
```
Comment (sashuk) [inline packages/picasso-rich-text-editor/src/plugins/ImagePlugin/ImagePluginButton.tsx:13]: How about removing focused from `useRTEPluginContext()` results? It seems like plugins should not care if the editor is focused, they only need a flag to disable button

```
 export const useRTEPluginContext = () => {
   const { disabled, focused } = useContext(RTEPluginContext)

   return {
-    disabled,
+   isDisabled: disabled || !focused
-    focused,
   }
}
```

Screenshot 2023-07-04 at 19 42 53

File: packages/picasso-rich-text-editor/src/plugins/ImagePlugin/ImagePluginModal.tsx [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+            value={files}
+            accept='image/png, image/jpeg'
+            onChange={upload}
+            onRemove={remove}
+            maxFiles={null}
+          />
+        </Form.Field>
+        <Form.Field>
+          <Form.Label>Alt text</Form.Label>
+          <Input
```
Comment (sashuk) [inline packages/picasso-rich-text-editor/src/plugins/ImagePlugin/ImagePluginModal.tsx:117]: Asked @TomasSlama about the design of the alt text input

File: packages/picasso-rich-text-editor/src/plugins/ImagePlugin/ImagePluginModal.tsx [hunk 6]
```diff
@@ -0,0 +1,133 @@
+import React, { useEffect, useState } from 'react'
```
Comment (sashuk) [inline packages/picasso-rich-text-editor/src/plugins/ImagePlugin/ImagePluginModal.tsx:1]: Asked @TomasSlama if is expected that the modal is not reset after the submit button is pressed, the modal closes, and the button is clicked again

**Review by sashuk — COMMENTED** [review]
Open question were clarified, looks good!

**Review by sashuk — APPROVED** [review]

**Pull request merged**